src/parquet/file/metadata-internal.cc

```diff
@@ -0,0 +1,650 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <vector>
+
+#include "parquet/file/metadata.h"
+#include "parquet/thrift/util.h"
+
+#define VALIDATE_COLUMN(X)                                               \
+  DCHECK(X < num_columns()) << "The RowGroup only has " << num_columns() \
+                            << "columns, requested column: " << X;
+
+namespace parquet {
+
+namespace metadata {
```

Comment:
Since the class names are fairly unambiguous, this extra level of namespace may be unnecessary (contrast this with the `parquet::schema::Node` class)